### DISCUSSION TITLE
No index.html is created in the coverage folder

### DISCUSSION THREAD

**Discussion by Ben-at-h** [vitest-dev/vitest#5418] [Q&A]
When trying to get test coverage to work, I have followed the guidance in the vitest documentation. However, no coverage report is created when I run command "vitest run --coverage". It seems that the UI is looking for an index.html within the coverage folder, but it only has JSON files within it e.g. coverage-40.json.

Am I correct there should be an index.html file within this directory, and if so any ideas why it's not being created?

**Comment by Ben-at-h** [comment]
I've noticed that they are in a .tmp folder. I think what is happening is vitest isn't completing the process of creating the coverage report.

It seems that the coverage report only shows when all tests are passing? Can I suggest this information is included into the documentation, as it took me a long time to work this out 😫

**Comment by cadenzah** [comment]
Same problem here, isn't there no way to change the directory which vitest-ui is looking at for coverage report? Or the directory where index.html and other assets are created should be able to be changed.

**Reply by AriPerkkio** [reply]
There's [`coverage.reportsDirectory`](https://vitest.dev/config/#coverage-reportsdirectory), see also [Coverage | Guide | Changing the Default Coverage Folder Location.
](https://vitest.dev/guide/coverage.html#changing-the-default-coverage-folder-location)

If you are running into the issue that this out-dated discussion originally reported, I would recommend to open new bug report.

**Reply by cadenzah** [reply]
For those who are interested in, I opened [a new issue](https://github.com/nuxt/test-utils/issues/869)
